simpleclient_logback/src/main/java/io/prometheus/client/logback/InstrumentedAppender.java

```diff
@@ -0,0 +1,84 @@
+package io.prometheus.client.logback;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.Appender;
+import ch.qos.logback.core.UnsynchronizedAppenderBase;
+import io.prometheus.client.CollectorRegistry;
+import io.prometheus.client.Counter;
+
+public class InstrumentedAppender extends UnsynchronizedAppenderBase<ILoggingEvent> {
+
+  public static final String COUNTER_NAME = "logback_appender_total";
+
+  private final CollectorRegistry registry;
+
+  private Counter counter;
+  private Counter.Child trace;
+  private Counter.Child debug;
+  private Counter.Child info;
+  private Counter.Child warn;
+  private Counter.Child error;
+
+  /**
+   * Create a new instrumented appender using the default registry.
+   */
+  public InstrumentedAppender() {
+    this(CollectorRegistry.defaultRegistry);
+  }
+
+  /**
+   * Create a new instrumented appender using the given registry.
+   *
+   * @param registry the collector registry
+   */
+  public InstrumentedAppender(CollectorRegistry registry) {
+    this.registry = registry;
+    setName(Appender.class.toString());
+  }
+
+  @Override
+  public void start() {
```

Comment:
I added an effective singleton check to both of the appender implementations.. However, I'm not a big fan of how its done or the need for it. It doesn't feel right. I read the existing log4j and logback appender implementations for things like files and sockets etc and they don't try to be clever and ensure anything is configured incorrectly (as two file or socket appenders pointing to the same file/socket).  Its by convention.

It would be good in the long run to have prometheus throw an exception or at least provide the ability to check if a metric is already registered. Accidentally being able to break and entire endpoint from scraping by adding a metric with the same name isn't feasible in a large system. Is there a ticket where I can vote for this to be fixed? A solution to that would also remove the need for the separate static boolean singletonFlag in this code.